### PR TITLE
Remove gogoproto options from gateway.proto

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

Remove gogoproto options from `gateway.proto` and `gateway_services.proto`. Please check `Notes for Reviewers` for details.

This is based off https://github.com/TheThingsNetwork/lorawan-stack/pull/4662 and is hence targeting the `issue/2798-gogo-gtw-identifiers` branch. Once that PR is merged, I'll rebase this on `v3.15`.

#### Changes

- Remove `gogoproto.nullable` and `gogoproto.embed` options from `gateway.proto` and `gateway_services.proto`
- Adapt interfaces and logic
- Add tests.

#### Testing

UT

##### Regressions

I think I've caught them mostly in UT.

#### Notes for Reviewers

- `PullGatewayConfigurator` doesn't seem to be used anywhere. I've added the interfaces for completeness nonetheless.
- The `gateway.version_ids` message was set as required `(validate.rules).message.required = true` but it's not. The `required` was effective only after removing the nullable option so I removed that as well.
- The `(gogoproto.nullable) = true` on `schedule_anytime_delay` had no effect.
- There's more than one way of handling the `GatewayAntennas` option. I've chosen the easiest one which requires the smallest amount of changes.
- This is a lot of manual work so please check your code owned files.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/gatewayserver/gatewayserver_test.go [hunk 1]
```diff
@@ -753,7 +758,7 @@ func TestGatewayServer(t *testing.T) {
 									UpdateLocation: false,
 									Up: &ttnpb.GatewayUp{
 										GatewayStatus: &ttnpb.GatewayStatus{
-											Time: time.Unix(424242, 0),
+											Time: timePtrUTC(time.Unix(424242, 0).UTC()),
```
Comment (adriansmares) [inline pkg/gatewayserver/gatewayserver_test.go]: Do you need to `UTC` twice ? The pointer function does it too.

File: pkg/gatewayserver/io/mqtt/format_protobufv2.go [hunk 2]
```diff
@@ -66,6 +66,11 @@ type protobufv2 struct {
 	topics.Layout
 }
 
+// TimePtr converts time to a pointer.
+func TimePtr(x time.Time) *time.Time {
+	return &x
```
Comment (adriansmares) [inline pkg/gatewayserver/io/mqtt/format_protobufv2.go:71]: Does this need `UTC` ? Maybe that's how we end up with these weirdly deformed timestamps of gateway times (not saying this call site necessarily, but _maybe_...)

**Review by adriansmares — APPROVED** [review]

File: pkg/gatewayserver/io/mqtt/format_protobufv2.go (hunk 2 above)
Comment (KrishnaIyer) [reply pkg/gatewayserver/io/mqtt/format_protobufv2.go:71]: Good question. The reason I added UTC in the other test was that when it was dereferenced by the assertions package, it takes local time. Let me check what the GS does in general with time and get back to you.

**Review by htdvisser — APPROVED** [review]
LGTM

File: pkg/gatewayserver/io/mqtt/format_protobufv2.go (hunk 2 above)
Comment (htdvisser) [reply pkg/gatewayserver/io/mqtt/format_protobufv2.go:71]: I'd say this is out of scope for this PR, but indeed good to check.

**Comment by KrishnaIyer** [comment]
@kschiffer: Could you please take a look at [this](https://github.com/TheThingsNetwork/lorawan-stack/pull/4674/checks?check_run_id=3731017663#step:9:238) test case?

The `GatewayAntennas` field of the `Gateway` message is now an array of pointers so that's where I think this assertion is failing. Cypress takes forever on my laptop so I'd appreciate it if you could take a look.

**Comment by htdvisser** [comment]
This one missed the deadline for v3.15.1, so moving it to v3.15.2.

**Pull request merged**